include/nbl/builtin/hlsl/spirv_intrinsics.hlsl

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_SPIRV_INTRINSICS_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SPIRV_INTRINSICS_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SPIRV_INTRINSICS_INCLUDED_
+
+#ifdef NBL_GL_KHR_shader_subgroup_shuffle_relative
+[[vk::ext_capability(/* GroupNonUniformShuffleRelative */ 66)]]
+void spirv_shuffle_cap(){}
+#endif
```

Comment:
ok this stuff will emit the SPIR-V for an extension whenever you include the header, you want to avoid this by breaking up the contents of `spirv_intrinsics.hlsl` into a directory called `spirv_intrinsics` with different `.hlsl` inside.

If we continue this pattern then we'll get quite the unholy mess where including the SPIR or GLSL header will enable all extensions we implement (without checking for things like shader stage compatibility)